scripts/ci_test_packages.py

```
@@ -0,0 +1,68 @@
+    return "internal/" + package.rsplit(marker, 1)[1]
+
+
+def select_packages(packages: list[str], shard: str) -> list[str]:
+    heavy_packages = set(HEAVY_SHARDS.values())
+    if shard == "rest":
+        return [pkg for pkg in packages if package_suffix(pkg) not in heavy_packages]
+
+    wanted = HEAVY_SHARDS[shard]
+    selected = [pkg for pkg in packages if package_suffix(pkg) == wanted]
+    if not selected:
+        raise ValueError(f"no packages matched shard {shard!r}")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)